projects/lights_control/src/lights_simple_peripherals.c

```diff
@@ -0,0 +1,28 @@
+#include "event_queue.h"
+#include "fsm.h"
+#include "status.h"
+
+#include "lights_events.h"
+#include "lights_simple_peripherals.h"
+
+static SimplePeripheralCallback prv_simple_callback;
+
+StatusCode lights_simple_peripherals_init(SimplePeripheralCallback cb) {
+  prv_simple_callback = cb;
+  return STATUS_CODE_OK;
+}
+
+StatusCode lights_simple_peripherals_process_event(Event e) {
+  switch (e.id) {
```

Comment:
Why this complexity if they are all treated the same? Why not use a range `e.id >= LIGHTS_HORN_EVENT && e.id <= LIGHTS_EVENT_BRAKES` or some such?

Reply:
How can I namespace my events without different types of events colliging though? If you remember, I was gonna have some sort of a pub-sub system where `main` is going to pop a raised event, and call all the modules with that event. Each module processes the event if it belongs to it, or ignores it... 

If I have, say `LIGHTS_EVENT_SIMPLE_*` and `LIGHTS_EVENT_*`, then if I raise an event with the `EVENT_ID` of  0, then that will get processed by both modules that subscribe to the `LIGHTS_EVENT_SIMPLE_0` and `LIGHTS_EVENT__0`. 

Comparing the event id within a range sounds good though.

Reply:
For namespacing see how `projects/charger/chaos/inc/chaos_events.h` works.